src/peft/tuners/adaption_prompt.py

```diff
@@ -163,10 +168,10 @@ def __init__(self, model, configs: Dict, adapter_name: str):
         # Whether the adapter is enabled.
         self._enabled = True
         self.forward = self.model.forward
-        self.add_adapter(adapter_name, configs[adapter_name])
+        self._add_adapter(adapter_name, configs[adapter_name])
         self._mark_only_adaption_prompts_as_trainable()
 
-    def add_adapter(self, adapter_name: str, config: AdaptionPromptConfig) -> None:
+    def _add_adapter(self, adapter_name: str, config: AdaptionPromptConfig) -> None:
```

Comment:
The recursion error occurs because in `PeftModel`, it should be `self.base_model.add_adapter(adapter_name, peft_config)`, otherwise the method calls itself. The reason why renaming works is because of the `__getattr__` of `PeftModel` delegates to `base_model` (this confusion is btw. a good case in point why `__getattr__` should usually not be overridden).

Apart from that, it should not be necessary to rename the method, I tried it out locally and it works with the old name.